.github/workflows/server_tests.yaml

```diff
@@ -0,0 +1,61 @@
+name: server-tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - '*'
+  workflow_dispatch:
+
+jobs:
+  server-test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+#        python-version: [3.7, 3.8, 3.9, 3.10]
+        python-version: [3.9]
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest httpx pytest_asyncio docker pyyaml
+        pip install -r requirements.txt
+
+    - name: Set up Docker SSH Private Key
+      run: |
+        mkdir -p $HOME/.ssh/runhouse/docker
+        echo "${{ secrets.DOCKER_SSH_KEY }}" > $HOME/.ssh/runhouse/docker/id_rsa
+        chmod 600 $HOME/.ssh/runhouse/docker/id_rsa
+
+    - name: Set up Docker SSH Public Key
+      run: |
+        echo "${{ secrets.DOCKER_SSH_PUBLIC_KEY }}" > $HOME/.ssh/runhouse/docker/id_rsa.pub
+        chmod 600 $HOME/.ssh/runhouse/docker/id_rsa.pub
+
+    - name: Save Runhouse config
+      env:
+        TEST_TOKEN: ${{ secrets.TEST_TOKEN }}
+        TEST_USERNAME: ${{ secrets.TEST_USERNAME }}
+      run: |
+        mkdir -p $HOME/.rh
+        python -c "import yaml; import os; config = {'token': os.environ['TEST_TOKEN'], 'username': os.environ['TEST_USERNAME'], 'default_folder': f\"/{os.environ['TEST_USERNAME']}\", 'default_keypair': '$HOME/.ssh/runhouse/docker/id_rsa'}; yaml.dump(config, open('$HOME/.rh/config.yaml', 'w'))"
```

Comment:
I would actually rather we put these at `~/.ssh/sky-key` so they get picked up by default and we don't need to set a default_keypair, and then we can simply run `runhouse login -y $TEST_TOKEN` here instead of manually dropping in the config yaml, which is flakier and less realistic.

Reply:
Also, there's a big problem with this. We test sharing by sharing a resource with test account. In this action the sharer and sharee would be the same. We probably need a second CI test account or something that we use to log in here, so it's sharing with Test account just like we are locally.